sql/core/src/main/scala/org/apache/spark/sql/parquet/ParquetFilters.scala

```diff
@@ -40,6 +42,25 @@ private[sql] object ParquetFilters {
       createFilter(filter)
     }.reduceOption(FilterApi.and).map(FilterCompat.get)
   }
+    
+  case class SetInFilter[T <: Comparable[T]](
+    hSet: Set[T]) extends UserDefinedPredicate[T] with Serializable {
+
+    override def keep(value: T): Boolean = {
+      if (value == null) {
+        return false
+      }
+      return hSet.contains(value)
```

Comment:
Oh, just saw the `hset` field in `InSet`, still prefer `valueSet` here though...